addons/im_livechat/static/src/legacy/public_livechat.js

```diff
@@ -186,7 +186,7 @@ var LivechatButton = Widget.extend({
                     channel_uuid: this._livechat.getUUID(),
                     page_history: history,
                 });
-            } else { // normal message
+            } else if (notification[1].id) { // normal message
```

Comment:
`notification` is an array containing the uuid of the livechat and then a """notification""" object. I use multiple emphasis here because the second arg can be of multiple forms.

Initially the bug was that this code considered all received notifications as a received message as soon as `._type` is not `history_command`. The thing is : it also receives typing notifications and they were handled as messages, quickfixed by checking an id is given which means `notification[1]` is a message object

I have no idea of naming as it can be multiple things ... so the only idea I have is
```js
const [ livechatUUID, notificationData ] = notification:
```

Reply:
Any better idea ?

Reply:
No space inside array bracket but otherwise suggestion looks fine. Just make sure it can always de-structure and doesn't crash sometimes.